File: AuthenticationServices/src/main/java/com/artauction/authenticationservices/Repositories/Interfaces/BuyerRepo.java
package com.artauction.authenticationservices.Repositories.Interfaces;

import com.artauction.authenticationservices.Entities.Buyer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface BuyerRepo extends JpaRepository<Buyer, Integer> {
}

File: Project/Backend/src/ArtAuction.Infrastructure/Repositories/BuyerRepo.cs
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace ArtAuction.Infrastructure.Repositories;

public class BuyerRepo : IBuyerRepo
{
    // Attributes
    private readonly AppDbContext _dbContext;
    
    // Constructor
    public BuyerRepo(AppDbContext dbContext, ISaveChanges saveChanges)
    {
        this._dbContext = dbContext;
    }
    
    // Methods
    public async Task<Buyer?> GetBuyerById(int buyerId)
    {
        return await _dbContext.Buyers.AsNoTracking().SingleOrDefaultAsync(buyer => buyer.Id == buyerId);
    }
}